pkg/render/fluentd.go

```diff
@@ -521,6 +589,29 @@ func (c *fluentdComponent) container() corev1.Container {
 	}, c.cfg.ESClusterConfig.ClusterName(), ElasticsearchLogCollectorUserSecret, c.cfg.ClusterDomain, c.cfg.OSType)
 }
 
+func (c *fluentdComponent) metricsService() *corev1.Service {
```

Comment:
Interesting perspective that I had not considered. Though I do not see a disadvantage for having the service when no monitoring is enabled. It might even be nice to have a dns name for curling some fluentd metrics for occasional checks or scripting.

It kind of makes you wonder where to stop. If there is not Manager CR, do we also want to remove the compliance server service, the es gateway service, and so on...?

Reply:
So why I don't think there's zero advantage here, and why I don't think this is completely different from  the compliance server service or gateway service, is because this _**is**_ useless on it's own. Prometheus doesn't use the service to get metrics, it uses the endpoints the are created for the service that contain the IPs for all the fluentd pods. If you use the service directly you will get metrics for a _**completely random**_ fluentd pod, so the service itself is useless for fluentd.

es gateway, compliance server, those are services for access compliance and es gateway, and it doesn't matter which pod it goes to (the point of a "load balancing" service).

Reply:
If there is a valid use-case for us to have the service when running prometheus, then there would probably be a valid use case for customers to have the service if they want to have another prometheus read fluentd. For this reason I do not think this service is different from the compliance server service.

We currently do not support disabling the metrics. I think if we had the option to disable metrics, it would make sense to also not create the service.

Right now, I think we can just keep it.

Reply:
> If there is a valid use-case for us to have the service when running prometheus, then there would probably be a valid use case for customers to have the service if they want to have another prometheus read fluentd.

I would say that if a user wants fluentd metrics for their own prometheus then they get to make the choice of how to gather it and they would add the service if needed. If pod monitors supported mTLS they could use a pod monitor, but either way if they're using a different prometheus they need to setup their own monitors since our monitors are for our prometheus.

Also to state, this service is not for fluentd, this service is for prometheus, fluentd's part in this is over once it exposes the metrics on it's pod (these metrics are pod specific not service specific so that's why I say it's part is over). It's prometheus that has the responsibility of gathering the metrics from those pods, and to get the IP's of the pods prometheus needs a service, but only because there is a limitation in the pod monitor as the pod monitor would be the appropriate avenue for this.

> We currently do not support disabling the metrics. I think if we had the option to disable metrics, it would make sense to also not create the service.

Don't we have a CR that we respond to for metrics? Regardless of this we should the appropriate components in the appropriate spot.